db-es-audit/src/find_granules.py

```
@@ -0,0 +1,442 @@
+    WHERE rn = 1
+    """
+
+    result = False
+    try:
+        with db_connection.cursor() as curr:
+            curr.execute(search_query, cid=concept_id)
+            result = curr.fetchone()[0]
+
+    except oracledb.Error as e:
+            logger.error(f"In has_granule_been_deleted Exception occured. {e}")
+            raise Exception(f"Failed to execute query: {search_query}\nerror: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)